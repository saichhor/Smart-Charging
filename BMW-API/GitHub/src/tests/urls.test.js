import BMWURLs, { getHost } from '../models/urls';

const URLs = new BMWURLs('eu');

describe('Testing URLS', () => {
    it('should return baseURL', () => {
        expect(URLs.getHost()).toBe('b2vapi.bmwgroup.com');
    });
    it('should return authURL', () => {
        expect(URLs.getAuthURL()).toBe('https://b2vapi.bmwgroup.com/gcdm/oauth/token');
    });
    it('should return vehicleStatusURL', () => {
        expect(URLs.getVehicleStatusURL('test')).toBe(
            'https://b2vapi.bmwgroup.com/webapi/v1/user/vehicles/test/status'
        );
    });
    it('should return getRemoteServiceStatusURL', () => {
        expect(URLs.getRemoteServiceStatusURL('test', 'test')).toBe(
            'https://b2vapi.bmwgroup.com/webapi/v1/user/vehicles/test/serviceExecutionStatus?serviceType=test'
        );
    });
    it('should return getRemoteServiceURL', () => {
        expect(URLs.getRemoteServiceURL('test')).toBe(
            'https://b2vapi.bmwgroup.com/webapi/v1/user/vehicles/test/executeService'
        );
    });
    it('should return getVehicleImage', () => {
        expect(URLs.getVehicleImage('test', 100, 100, 'test')).toBe(
            'https://b2vapi.bmwgroup.com/webapi/v1/user/vehicles/test/image?width=100&height=100&view=test'
        );
    });
    it('should throw when no region', () => {
        expect(getHost).toThrowErrorMatchingSnapshot();
    });
});
