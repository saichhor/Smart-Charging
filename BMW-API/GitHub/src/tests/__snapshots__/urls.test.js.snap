// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing URLS should throw when no region 1`] = `"Unsupported region. Supported values are us|eu|cn"`;
